### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by achoum**
- Simple benchmark to test individually operators.
- Rename current benchmarking logic into "profiling" (this is more accurate).
- Also, fix an issue with the sampling operator.

Current benchmark output (on my machine):

```
================================================================
Name                              Wall time (s)    CPU time (s)
================================================================
from_dataframe:s:10000_numidx:0_numidxval:20_idx:int       0.00180       0.00179
from_dataframe:s:10000_numidx:0_numidxval:20_idx:str       0.00189       0.00189
from_dataframe:s:10000_numidx:1_numidxval:20_idx:int       0.00709       0.00709
from_dataframe:s:10000_numidx:1_numidxval:20_idx:str       0.01272       0.01272
from_dataframe:s:10000_numidx:3_numidxval:20_idx:int       1.17666       1.17665
from_dataframe:s:10000_numidx:3_numidxval:20_idx:str       1.86716       1.86716
from_dataframe:s:10000_numidx:5_numidxval:20_idx:int       2.16228       2.16226
from_dataframe:s:10000_numidx:5_numidxval:20_idx:str       3.92594       3.92591
----------------------------------------------------------------
simple_moving_average:100            0.00112       0.00112
simple_moving_average:10000          0.00168       0.00168
simple_moving_average:1000000        0.00880       0.00880
----------------------------------------------------------------
select_and_glue:100                  0.00101       0.00101
select_and_glue:10000                0.00097       0.00097
select_and_glue:1000000              0.00103       0.00103
----------------------------------------------------------------
calendar_day_of_month:100            0.00032       0.00032
calendar_day_of_month:10000          0.00593       0.00593
calendar_day_of_month:1000000        0.57562       0.57557
----------------------------------------------------------------
sample:e100_s100                     0.00118       0.00117
sample:e100_s10000                   0.00272       0.00272
sample:e100_s1000000                 0.00577       0.00577
sample:e10000_s100                   0.00131       0.00131
sample:e10000_s10000                 0.00148       0.00148
sample:e10000_s1000000               0.00710       0.00710
sample:e1000000_s100                 0.00174       0.00174
sample:e1000000_s10000               0.00260       0.00260
sample:e1000000_s1000000             0.01008       0.01008
----------------------------------------------------------------
propagate:100                        0.00082       0.00082
propagate:10000                      0.00816       0.00816
propagate:1000000                    0.61049       0.61049
================================================================
```

### CODE REVIEW

**Review by getchebarne — APPROVED** [review]
Really awesome tool 🚀  For the future, we can add command line arguments to control which benchmarks to run and their parameters.

File: benchmark/benchmark_time.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+"""
+Benchmark Python API.
+"""
+
+import time
+import numpy as np
+import temporian as tp
+from temporian.implementation.numpy.data.event import NumpyEvent
+import pandas as pd
+from typing import List, NamedTuple, Union
```
[the remark this answers is not in the document]
Comment (achoum) [reply benchmark/benchmark_time.py:24]: Good point.
*Done

File: benchmark/benchmark_time.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+
+def _build_toy_dataset(n, data_prefix="", data2_is_categorical_integer=False):
```
[the remark this answers is not in the document]
Comment (achoum) [reply benchmark/benchmark_time.py:27]: *Done

**Comment by achoum** [comment]
Thanks

**Pull request merged**